Solutions/Marain.Workflow.Abstractions/Marain/Workflows/IWorkflowInstanceChangeLogReader.cs

```diff
@@ -0,0 +1,24 @@
+﻿// <copyright file="IWorkflowInstanceChangeLogReader.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Workflows
+{
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Defines the interface for reading a log of the history of workflow instance changes.
+    /// </summary>
+    public interface IWorkflowInstanceChangeLogReader
+    {
+        /// <summary>
+        /// Gets a page of log entries for a particular workflow instance ID.
+        /// </summary>
+        /// <param name="workflowInstanceId">The ID of the workflow instance for which to get log entries.</param>
+        /// <param name="startingTimestamp">The initial sequence number with which to start enumerating entries.</param>
```

Comment:
If it's a timestamp, but also an int, shouldn't there be some explanation of how to go from a `DateTime` (or similar) to the `int` expected by this interface?

Reply:
I am considering the language and going back to being a non-contiguous sequence number. It will happen to be derived from a timestamp, but isn't one.

Reply:
I am sticking with timestamp but at millisecond resolution. Note that this does not resolve the "you have to be prepared for things having the same timestamp or being out of order because of clock drift/reset."